### PR TITLE
refactor: Event's content::RenderFrameHost is no longer needed to reply to synchronous sendSync IPC messages. 

### PR DESCRIPTION
**Pull request by pi1024e**
#### Description of Change
 Having mojo callbacks means we do not need Events to have content::WebContentsObserver anymore. 
#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x ] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes
Notes:  Replaced SetSenderAndMessage with SetCallBack because we do not need Event to reply to IPC messages.

### CODE REVIEW

File: lib/browser/api/protocol.ts [hunk 1]
```diff
@@ -6,19 +6,21 @@ const protocol = process.electronBinding('protocol')
...
-    if (!Object.getPrototypeOf(protocol).hasOwnProperty(property)) return
-
-    // Returning a native function directly would throw error.
-    return (...args: any[]) => (protocol[property as keyof Electron.Protocol] as Function)(...args)
+    if (app.isReady()) {
+      const protocol = session.defaultSession!.protocol
+      if (Object.getPrototypeOf(protocol).hasOwnProperty(property)) {
+        // Returning a native function directly would throw error.
+        return (...args: any[]) => (protocol[property as keyof Electron.Protocol] as Function)(...args)
+      }
```
Comment (nornagon) [inline lib/browser/api/protocol.ts]: The refactors in this file are unrelated, please revert them.

File: shell/browser/api/event.h [hunk 2]
```diff
@@ -26,9 +25,8 @@ class Event : public Wrappable<Event>, public content::WebContentsObserver {
   static void BuildPrototype(v8::Isolate* isolate,
                              v8::Local<v8::FunctionTemplate> prototype);
 
-  // Pass the sender and message to be replied.
-  void SetSenderAndMessage(content::RenderFrameHost* sender,
-                           base::Optional<MessageSyncCallback> callback);
+  // Pass the callback to be invokes
```
Comment (nornagon) [inline shell/browser/api/event.h]:
```suggestion
  // Set the callback to be invoked when replying to the message.
```

**Comment by nornagon** [comment]
Hm... actually it looks like this is already in an open PR as https://github.com/electron/electron/pull/19251

**Comment by pi1024e** [comment]
Did those changes!

**Comment by nornagon** [comment]
Closing as this is a dupe of #19251.
